client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,317 @@
+import { $, $a, removeFromDOM, asArray, escapeID } from '../domhelpers.js';
+import { Widget } from './widget.js';
+
+class Scoreboard extends Widget {
+  constructor(object, surface) {
+    super(object, surface);
+
+    this.addDefaults({
+      movable: true,
+      width: 300,
+      height: 200,
+      layer: -1,
+      typeClasses: 'widget scoreboard',
+      playersInColumns: true,
+      rounds: null,
+      roundLabel: 'Round',
+      scoreProperty: 'score',
+      seats: null,
+      showAllRounds: false,
+      showAllSeats: false,
+      showPlayerColors: true,
+      showTotals: true,
+      sortField: 'index',
+      sortAscending: true,
+      currentRound: null,
+      autosizeColumns: true
+    });
+  }
+
+  applyDeltaToDOM(delta) {
+    if(delta) // Don't call super unless this is a real delta to the scoreboard rather than from a seat
+      super.applyDeltaToDOM(delta);
+    this.tableCreate(this.getIncludedSeats())
+  }
+
+  classes() {
+    let className = super.classes();
+
+    if(this.get('autosizeColumns') != '')
+      className += ' equalWidth';
+
+    return className;
+  }
+
+  classesProperties() {
+    const p = super.classesProperties();
+    p.push('autosizeColumns');
+    return p;
+  }
+
+  get(property) {
+    if(property != '_totals')
+      return super.get(property)
+    else if(!this.get('showTotals'))
+      return [];
+    else {
```

Comment:
After return statements you don't need to use else. I find reducing the number of indented blocks to keep track of helps make code more readable.

Reply:
Overtaken by other events.